backend/utilities/movieAPIUtilities.js

```diff
@@ -58,22 +58,46 @@ const fetchTorrentData = async (imdbCode) => {
   return movieData;
 };
 
+const getParsedInfo = (res) => {
+  if (res && res.data && res.data.data && res.data.data.movies && res.data.data.movies[0]) {
+    const movie = res.data.data.movies[0];
+    return {
+      Title: movie.title || '',
+      imdbRating: movie.rating || '',
+      Year: movie.year || '',
+      Genre: movie.genres.join(', '),
+      Plot: movie.description_full || '',
+      Runtime: '',
+      Director: '',
+      Actors: '',
+    };
+  }
+  throw notFoundError();
+};
+
 const fetchMovieInfo = async (imdbCode) => {
-  const res = await axios.get(`http://www.omdbapi.com/?i=${imdbCode}&apikey=${OMDB_KEY}`);
-  const movieData = res.data;
-  if (res.status !== 200 || movieData.Response === 'False') throw notFoundError(); // omdbapi is not very restful
+  let movieData;
+  let res;
+  try {
+    res = await axios.get(`http://www.omdbapi.com/?i=${imdbCode}&apikey=${OMDB_KEY}`);
+    movieData = res.data;
+    if (res.status !== 200 || movieData.Response === 'False') throw notFoundError();
+  } catch (e) {
+    res = await axios.get(`${TORRENT_API}?query_term=${imdbCode}`);
+    movieData = getParsedInfo(res);
```

Comment:
I'm wondering what happens if the torrent api call in the catch block goes wrong?

Reply:
It will throw error, that we will catch in our middleware. frontend will receive our custom error, and it handles it there.

Reply:
Ah of course! I forgot how brilliant the backend is :D